src/pages/RepoPage/CoverageTab/subroute/FileExplorer/CodeTreeTable/CodeTreeTable.tsx

```diff
@@ -0,0 +1,217 @@
+import {
+  createColumnHelper,
+  flexRender,
+  getCoreRowModel,
+  useReactTable,
+} from '@tanstack/react-table'
+import cs from 'classnames'
+import { useMemo, useState } from 'react'
+
+import { OrderingDirection } from 'services/repos'
+import { Row } from 'shared/ContentsTable/utils'
+import Icon from 'ui/Icon'
+
+import { useRepoBranchContentsTable } from '../hooks'
+import { Loader, RepoContentsResult } from '../shared'
+
+const columnHelper = createColumnHelper<Row>()
+
+function getOrderingDirection(sorting: Array<{ id: string; desc: boolean }>) {
+  const state = sorting.at(0)
+
+  if (state) {
+    const direction = state?.desc
+      ? OrderingDirection.DESC
+      : OrderingDirection.ASC
+    let ordering = undefined
+    if (state.id === 'name') {
+      ordering = 'NAME'
+    }
+
+    if (state.id === 'percentCovered') {
+      ordering = 'COVERAGE'
+    }
+
+    if (state.id === 'hits') {
+      ordering = 'HITS'
+    }
+
+    if (state.id === 'misses') {
+      ordering = 'MISSES'
+    }
+
+    if (state.id === 'partials') {
+      ordering = 'PARTIALS'
+    }
+
+    if (state.id === 'lines') {
+      ordering = 'LINES'
+    }
+
+    return { direction, ordering }
+  }
+
+  return undefined
+}
+
+export const getBaseColumns = () => {
+  const baseColumns = [
+    columnHelper.accessor('name', {
+      id: 'name',
```

Comment:
Just a note for the future, no action required here, but with the column helper you shouldn't be required to assign an id as the helper should do it for you iirc.